src/users/users.service.ts

```diff
@@ -1,39 +1,56 @@
 import crypto from 'crypto';
 import { AES } from 'crypto-js';
 import { Secret } from 'otpauth';
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
 import { PrismaService } from 'src/prisma/prisma.service';
-import { User } from './interfaces/user.interface';
+import { UserDto } from './interfaces/user.interface';
+import { RegisterUserDto } from './dto/register-user.dto';
+import { LoginUserDto } from './dto/login-user.dto';
+import { NonUniqueException } from 'src/infrastructure/non-unique-exception';
 
 @Injectable()
 export class UsersService {
-  private readonly users: User[] = [];
-
   constructor(private readonly prisma: PrismaService) {}
 
-  create(): User {
+  async create(registerUserDto: RegisterUserDto): Promise<UserDto> {
     const secret = new Secret({ size: 16 }).b32;
     const key = crypto.randomBytes(16).toString('hex');
     const encryptedSecret = AES.encrypt(secret, key).toString();
 
-    const user: User = {
-      id: Math.random().toString(36).substring(7),
-      name: 'zero-based',
-      email: 'zerobasedteam@gmail.com',
-      secret,
-      key,
-    };
+    let user: UserDto;
+    try {
+      user = await this.prisma.user.create({
+        data: {
+          ...registerUserDto,
+          secret: encryptedSecret,
+        },
+      });
+    } catch (err) {
+      throw new NonUniqueException(err);
```

Comment:
We should specify which field(s) are violating the uniqueness constraints using `IFieldValidationError`.